res/controllers/Roland_DJ-505-scripts.js

```diff
@@ -371,6 +371,9 @@ DJ505.crossfader.setReverse = function(channel, control, value, _status, _group)
 DJ505.setChannelInput = function(channel, control, value, _status, _group) {
     const number = (channel === 0x00) ? 0 : 1;
     const channelgroup = "[Channel" + (number + 1) + "]";
+    if (engine.getValue(channelgroup, "input_configured") === 0) {
+        throw "Configuration Error: Mapping expects audio inputs to be configured! please refer to https://manual.mixxx.org/latest/hardware/controllers/roland_dj_505#audio-setup";
+    }
```

Comment:
IIUC this should only be thrown if the input value is 0x01 or 0x02, right?
I.e. when the 505 reports the controls states on startup, and when the user flicks the switch.

In that regard the manual is a bit misleading, users may be confused by
> You should assign the Vinyl Control input channels even if you do not intend to use timecode vinyl. 

because if they don't intend to ever use VC, or use VC with another sound card or whatever, there is no **need** to configure the inputs.

Bottom line: the current message is okay because it matches the manual, but IMHO the manual should be changed.

We could still rephrase it a bit:
```suggestion
    if (engine.getValue(channelgroup, "input_configured") === 0) {
        throw "Configuration Error: a PC/Line/Phono switch has been set to Line or Phono. In order to use these modes audio inputs to be configured accordingly! Please refer to https://manual.mixxx.org/latest/hardware/controllers/roland_dj_505#audio-setup";
    }
```

Reply:
> In that regard the manual is a bit misleading, users may be confused by
>
>> You should assign the Vinyl Control input channels even if you do not intend to use timecode vinyl.
>
> because if they don't intend to ever use VC, or use VC with another sound card or whatever, there is no need to configure the inputs.

At the time of writing this, I was not using Vinyl Control (= DVS) at all, but I did include some actual vinyl records into my mixes. So even though I did not intend to use vinyl control, I still needed to configure these inputs for passing though non-timecode records. Maybe we should rename the input channels to "Phono" in the preferences ;)

Reply:
I like the rephrased version a lot more - and also the renaming in the preferences would be good